enos/modules/upgrade_servers/main.tf

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+terraform {
+  required_providers {
+    enos = {
+      source = "registry.terraform.io/hashicorp-forge/enos"
+    }
+  }
+}
+
+resource "random_pet" "upgrade" {
+}
+
+resource "enos_local_exec" "wait_for_nomad_api" {
+  environment = {
+    NOMAD_ADDR        = var.nomad_addr
+    NOMAD_CACERT      = var.ca_file
+    NOMAD_CLIENT_CERT = var.cert_file
+    NOMAD_CLIENT_KEY  = var.key_file
+    NOMAD_TOKEN       = var.nomad_token
+
+  }
+
+  scripts = [abspath("${path.module}/scripts/wait_for_nomad_api.sh")]
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//                    Upgrading the first server
+////////////////////////////////////////////////////////////////////////////////
+// Taking a snapshot forces the cluster to store a new snapshot that will be 
+// used to restore the cluster after the restart, because it will be the most 
+// recent available, the resulting file wont be used..
+resource "enos_local_exec" "take_first_cluster_snapshot" {
+  depends_on = [enos_local_exec.wait_for_nomad_api]
+
+  environment = {
+    NOMAD_ADDR        = var.nomad_addr
+    NOMAD_CACERT      = var.ca_file
+    NOMAD_CLIENT_CERT = var.cert_file
+    NOMAD_CLIENT_KEY  = var.key_file
+    NOMAD_TOKEN       = var.nomad_token
+  }
+
+  inline = [
+    "nomad operator snapshot save ${random_pet.upgrade.id}-0.snap",
+  ]
```

Comment:
Without the [`-stale`](https://developer.hashicorp.com/nomad/docs/commands/operator/snapshot/save#stale) flag, this is always getting forwarded to the leader.

Reply:
Let me test with the stale flag and specifying the server address